package-examples/kustomize-pkg/README.md

```diff
@@ -0,0 +1,163 @@
+# kustomize-pkg
+
+This is a simple package that shows how kpt packages can be used instead of remote bases in kustomize.  That allows you to take advantage of kpt's rebase and local, in-place edits.
+
+## Steps
+
+1. [Fetch the package](#fetch-the-package)
+2. [View the package contents](#view-the-package-contents)
+3. [kustomize the config](#kustomize-the-config)
+4. [Render the kustomization](#render-the-kustomization)
+5. [Apply the package](#apply-the-package)
+
+### Fetch the package
+
+Get the example package on to local using `kpt pkg get`. Note that this package is for this example, wihin that package we are also using the nginx sub-package which is an alternative to having a remote base. 
+
+```shell
+$ kpt pkg get https://github.com/GoogleContainerTools/kpt.git/package-examples/kustomize-pkg
+
+fetching package /package-examples/kustomize-pkg from https://github.com/GoogleContainerTools/kpt to kustomize-pkg
+```
+
+
+### View the package contents
+
+List the package contents in a tree structure.
+
+```shell
+$ kpt pkg tree kustomize-pkg/
+
+PKG: kustomize-pkg
+├── [Kptfile]  Kptfile kustomize-pkg
+├── PKG: nginx
+│   ├── [Kptfile]  Kptfile nginx
+│   ├── [deployment.yaml]  Deployment my-nginx
+│   └── [svc.yaml]  Service my-nginx-svc
+├── dev
+│   ├── [kustomization.yaml]  Kustomization 
+│   └── [pass-patch.yaml]  Deployment deployment-patch
+└── prod
+    ├── [kustomization.yaml]  Kustomization 
+    └── [pass-patch.yaml]  Deployment deployment-patch
+```
+
+### kustomize the config
+
+We recommend that you keep the kustomize instructions to rendering only such as adding a namespace, transforming or applying a patch.  The second you mix kpt packages and remote bases you will be missing out on a big advantage of having a guaranteed stable base.
+
+You can edit the patches and kustomize files we have created in the overlay folders.
+
+```yaml
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+resources:
+- ../../bases/nginx
```

Comment:
This structure is probably ok for an example, but I think the recommended way to do this would be to have each env have separate git references. The structure in this example makes it a bit hard to make changes in the base and then do progressive rollouts into different environments.

Reply:
What would be the layout of a solution like that?  Every overlay then gets a separate base?  Why create an overlay then?  We can just have the final config in every folder.  I haven't seen a pattern like that, even though I am not opposed to showing it.